src/backend/marsha/core/lti/__init__.py

```diff
@@ -1,25 +1,30 @@
 """LTI module that supports LTI 1.0."""
+from logging import getLogger
 import re
-from urllib.parse import urlparse
+from urllib.parse import unquote, urlparse
 
 from django.conf import settings
 from django.core.exceptions import ImproperlyConfigured
 from django.http.request import validate_host
 from django.utils.datastructures import MultiValueDictKeyError
 
+from oauthlib import oauth1
 from pylti.common import LTIException, verify_request_common
 
 from ..models import ConsumerSite
 from ..models.account import ADMINISTRATOR, INSTRUCTOR, LTI_ROLES, STUDENT, LTIPassport
 
 
+logger = getLogger(__name__)
+
+
 class LTI:
     """The LTI object abstracts an LTI launch request.
 
     It provides properties and methods to inspect the launch request.
     """
 
-    def __init__(self, request, resource_id):
+    def __init__(self, request, resource_id=None):
```

Comment:
I don't understand why you need to change this.

Reply:
Because a LTI ContentItemSelectionRequest isn't related to a resource.